integration-templates/google-calendar/google-calendar-events.ts

```diff
@@ -0,0 +1,28 @@
+import type { NangoSync, GoogleCalendarEvent } from './models';
+
+export default async function fetchData(nango: NangoSync): Promise<void> {
+    const oneMonthAgo = new Date();
+    oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);
+    oneMonthAgo.setHours(0, 0, 0, 0);
+    const timeMin = oneMonthAgo.toISOString();
+
+    let batch: GoogleCalendarEvent[] = [];
+    const batchSize = 100;
+    const maxResults = batchSize.toString();
+
+    const endpoint = `calendar/v3/calendars/primary/events?timeMin=${encodeURIComponent(timeMin)}`;
+
+    for await (const eventPage of nango.paginate<GoogleCalendarEvent>({ endpoint, params: { maxResults } })) {
+        for (const event of eventPage) {
+            batch.push(event);
+
+            if (batch.length === batchSize) {
+                await nango.batchSave<GoogleCalendarEvent>(batch, 'GoogleCalendarEvent');
+            }
+        }
+    }
+
+    if (batch.length > 0) {
+        await nango.batchSave<GoogleCalendarEvent>(batch, 'GoogleCalendarEvent');
+    }
```

Comment:
why do we need the intermediary `batch` array. can we not batchSave the result of paginate directly?

```
for await (const eventPage of nango.paginate<GoogleCalendarEvent>({ endpoint, params: { maxResults } })) {
   await nango.batchSave<GoogleCalendarEvent>(eventPage, 'GoogleCalendarEvent');
}
```

Reply:
good point! 

This was logic copied over from other sync scripts where there is some processing of the data, but in this case thats's not needed. Updated with https://github.com/NangoHQ/nango/pull/1269/commits/f109e782d52f15298e2967223aa076c5d7965b01